conjure-undertow-processor/src/main/java/com/palantir/conjure/java/undertow/processor/generate/ConjureUndertowEndpointsGenerator.java

```diff
@@ -629,15 +634,37 @@ public CodeBlock cookie(
             }
 
             @Override
-            public CodeBlock authCookie(String _variableName, String _cookieName, String deserializerFieldName) {
-                return CodeBlock.of("this.$N.deserialize($N)", deserializerFieldName, EXCHANGE_NAME);
+            public CodeBlock authCookie(String _variableName, String cookieName) {
+                return CodeBlock.of("this.$N.auth().cookie($N, $S)", RUNTIME_NAME, EXCHANGE_NAME, cookieName);
+            }
+
+            @Override
+            public CodeBlock optionalAuthCookie(String _variableName, String cookieName) {
+                return CodeBlock.of(
+                        "$2N.getRequestCookies().containsKey($3S) "
+                                + "? Optional.of(this.$1N.auth().cookie($2N, $3S)) "
+                                + ": Optional.empty()",
+                        RUNTIME_NAME,
+                        EXCHANGE_NAME,
+                        cookieName);
             }
 
             @Override
             public CodeBlock authHeader(String _variableName) {
                 return CodeBlock.of("this.$N.auth().header($N)", RUNTIME_NAME, EXCHANGE_NAME);
             }
 
+            @Override
+            public CodeBlock optionalAuthHeader(String _variableName) {
+                return CodeBlock.of(
+                        "$2N.getRequestHeaders().contains($3T.AUTHORIZATION) "
+                                + "? Optional.of(this.$1N.auth().header($2N)) "
+                                + ": Optional.empty()",
```

Comment:
This is a little gross, but otherwise we have to add new methods to `AuthorizationExtractor`. Doing so is difficult because this is a public interface that is used outside this repo, so this has some non-trivial ABI concerns.

We can always improve this in the future if we want do.